### PR TITLE
fix: for #6374 to delete non-empty directories

### PR DESCRIPTION
**Pull request by sellersj**
## Description of Change
Walk the files to delete using just java.

## Have test cases been added to cover the new functionality?
yes

### CODE REVIEW

**Review by jeremylong — APPROVED** [review]
LGTM

**Comment by jeremylong** [comment]
Thank you for reporting and **fixing** the issue. Really appreciate this.

File: utils/src/main/java/org/owasp/dependencycheck/utils/FileUtils.java [hunk 1]
```diff
@@ -91,7 +91,10 @@ public static boolean delete(@Nullable File file) {
         }
 
         try {
-            Files.delete(file.toPath());
+            Files.walk(file.toPath())
```
Comment (christian-ertl) [inline utils/src/main/java/org/owasp/dependencycheck/utils/FileUtils.java:94]: The stream returned by Files.walk should be closed.

Comment (jeremylong) [reply utils/src/main/java/org/owasp/dependencycheck/utils/FileUtils.java:94]: Interesting... most streams don't need to be closed like this; as documented in the Stream javadoc. Hadn't thought about this possibly being tied to an IO channel. Turns out this isn't uncommon: https://sourcegraph.com/search?q=context:global+content:%22try+%28Stream%3CPath%3E+paths+%3D+Files.walk%28%22+lang:java&patternType=standard&sm=1&groupBy=repo

Fixed with https://github.com/jeremylong/DependencyCheck/pull/6382

Thanks!

**Pull request merged**